File: CustomFonts/customFontsJava/src/main/java/aneil/customfontsjava/customcomponents/CFTextView.java
package aneil.customfontsjava.customcomponents;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.res.TypedArray;
import android.support.v7.widget.AppCompatTextView;
import android.util.AttributeSet;

import aneil.customfontsjava.R;
import aneil.customfontsjava.utils.FontManager;


/**
 * Created by neil.agarwal on 26/7/2018.
 * Custom implementation of Textview to adhere to custom font typeface
 */
public class CFTextView extends AppCompatTextView {

    public CFTextView(Context context) {
        super(context);
        setCustomFont(context, null);
    }

    public CFTextView(Context context, AttributeSet attrs) {
        super(context, attrs);
        setCustomFont(context, attrs);
    }

    public CFTextView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        setCustomFont(context, attrs);
    }

    /**
     * Sets custom font to textview
     *
     * @param context
     */
    @SuppressLint("CustomViewStyleable")
    private void setCustomFont(Context context, AttributeSet attrs) {
        String fontType = FontManager.MONTSERRAT_BLACK;
        if (null != attrs) {
            TypedArray array = context.obtainStyledAttributes(attrs, R.styleable.cfjView);
            int fontTypeInt = array.getInt(R.styleable.cfjView_cfFontStyle, CFFontStyle.MONTSERRAT_BLACK.getIndex());
            array.recycle();
            fontType = CFFontStyle.getORFontStyle(fontTypeInt).getFontName();
        }
        setTypeface(FontManager.getTypeface(context, fontType));
    }
}

File: CustomFonts/customFontsKt/src/main/java/aneil/customfonts/customcomponents/CFTextView.kt
package aneil.customfonts.customcomponents

import android.annotation.SuppressLint
import android.content.Context
import android.support.v7.widget.AppCompatTextView
import android.util.AttributeSet
import aneil.customfonts.R
import aneil.customfonts.utils.FontManager

class CFTextView : AppCompatTextView {
    constructor(context: Context?) : super(context) {
        setCustomFont(context, null)
    }

    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs) {
        setCustomFont(context, attrs)
    }

    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        setCustomFont(context, attrs)
    }


    /**
     * Sets custom font to textview
     *
     * @param context
     */
    @SuppressLint("CustomViewStyleable")
    private fun setCustomFont(context: Context?, attrs: AttributeSet?) {
        var fontType = FontManager.GOTHAMSSM_BLACK
        if (null != attrs) {
            val array = context?.obtainStyledAttributes(attrs, R.styleable.cfView)
            val fontTypeInt = array?.getInt(R.styleable.cfView_cfFontStyle, CFFontStyle.GOTHAMSSM_BLACK.index)
            array?.recycle()
            fontType = CFFontStyle.getORFontStyle(fontTypeInt).fontName
        }
        typeface = FontManager.getTypeface(context, fontType)
    }
}